peer/brontide.go

```diff
@@ -2105,52 +2115,94 @@ func (p *Brontide) queueHandler() {
 	lazyMsgs := list.New()
 
 	for {
-		// Examine the front of the priority queue, if it is empty check
-		// the low priority queue.
+		// Process the priority queue first. We will not accept any new
+		// messages until this queue is empty.
 		elem := priorityMsgs.Front()
-		if elem == nil {
-			elem = lazyMsgs.Front()
-		}
 
+		// Found an item, we will write it to lnwire.
 		if elem != nil {
-			front := elem.Value.(outgoingMsg)
+			front, ok := elem.Value.(outgoingMsg)
+			if !ok {
+				p.log.Errorf("Unknown message type found in "+
+					"priority queue: %v", elem)
+			}
+
+			p.log.Tracef("Sending priority msg to sendQueue: %s",
+				spew.Sdump(front))
 
-			// There's an element on the queue, try adding
-			// it to the sendQueue. We also watch for
-			// messages on the outgoingQueue, in case the
-			// writeHandler cannot accept messages on the
-			// sendQueue.
+			// There's an element on the queue, try adding it to
+			// the `sendQueue`.
 			select {
 			case p.sendQueue <- front:
-				if front.priority {
-					priorityMsgs.Remove(elem)
-				} else {
-					lazyMsgs.Remove(elem)
-				}
-			case msg := <-p.outgoingQueue:
-				if msg.priority {
-					priorityMsgs.PushBack(msg)
-				} else {
-					lazyMsgs.PushBack(msg)
-				}
+				// We know the elem must from priority queue.
+				priorityMsgs.Remove(elem)
+
+				// Once it's sent, we'll continue to the next
+				// iteration. This means we will try to send
+				// messages again if the queue is not empty,
+				// and we won't accept new messages until it's
+				// empty.
+				continue
```

Comment:
So for a given priorityMsg queue you first drain it before new messages get accepted, even priority messages?

Reply:
yeah i want the old priority messages to get processed first.